### PR TITLE
Update samples and tests to target netcoreapp2.1

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: test/Directory.Build.props [hunk 1]
```diff
@@ -2,6 +2,13 @@
   <Import Project="..\Directory.Build.props" />
 
   <PropertyGroup>
+    <DeveloperBuildTestTfms>netcoreapp2.1</DeveloperBuildTestTfms>
+    <StandardTestTfms>$(DeveloperBuildTestTfms)</StandardTestTfms>
+    <StandardTestTfms Condition=" '$(DeveloperBuild)' != 'true' ">netcoreapp2.1;netcoreapp2.0</StandardTestTfms>
+    <StandardTestTfms Condition=" '$(DeveloperBuild)' != 'true' AND '$(CoreOnly)' != 'TRUE' AND '$(OS)' == 'Windows_NT' ">$(StandardTestTfms);net461</StandardTestTfms>
```
Comment (smitpatel) [inline test/Directory.Build.props]: CoreOnly!= true in small caps?

Comment (pranavkm) [reply test/Directory.Build.props]: Sure.

Comment (bricelam) [reply test/Directory.Build.props]: `And` in pascal case? :trollface:

Comment (bricelam) [reply test/Directory.Build.props]: ...and no padding in conditions, please. 🤓

Comment (bricelam) [reply test/Directory.Build.props]: (Just kidding, I could care less how people format their code)

Comment (smitpatel) [reply test/Directory.Build.props]: I asked because I am not sure if the equality is case-sensitive or not.

Comment (bricelam) [reply test/Directory.Build.props]: Yeah, MSBuild don't care. I personally prefer `True` since it reminds me of QBASIC 😍

Comment (pranavkm) [reply test/Directory.Build.props]: MSBuild's string comparisons aren't case sensitive. I'll change it to `True`

**Comment by smitpatel** [comment]
Bad merge 😮

**Comment by pranavkm** [comment]
@smitpatel did I mess up something?

**Comment by smitpatel** [comment]
Commit which went in is good. But we like purple PR rather than red ones 😉 
In UI GitHub makes it easy to go to PR from commit to see discussions.

**Pull request closed without merging**